### PR TITLE
Added support for send simulate, changed recv-loop

### PR DESCRIPTION
**Pull request by TimGremalm**
Added the new function send_simulate.
irsend (the LIRC tool) have a mode called simulate where it could simulate received commands. This is useful for sending commands to LIRC.
The syntax looks like this:
    r = lirc_command_init(&cmd, "SIMULATE %016x %02x %s %s\n", scancode, repeat, keysym, remote);

Also changed the recv-loop to look for "BEGIN" in the whole data-string.
The first characters in data isn't always "BEGIN", it can contain newlines and other messages.

### CODE REVIEW

File: lircsend.py [hunk 1]
```diff
@@ -163,6 +163,11 @@ def list_remote_codes(self, remote):
 			return command_dict
 		return None
 
+	def send_simulate(self, scancode, repeat, keysym, remote):
+		command = "SIMULATE {0:0>16x} {1:0>2d} {2:s} {3:s}".format(scancode, repeat, keysym, remote)
+		lirc_result = self._send_packet(command)
+		return lirc_result
```
Comment (prcjac) [inline lircsend.py]: For code consistency can we inline the `lirc_result` variable so it reads `return self._send_packet(command)`

**Comment by prcjac** [comment]
Hi Tim,

Before I accept the pull request could you address the comment on the code above.

Thanks a lot!

Peter

File: lircsend.py (hunk 1 above)
Comment (TimGremalm) [inline lircsend.py]: Okey. Will fix that, thanks!

Comment (TimGremalm) [inline lircsend.py]: I've now updated my branch.

**Pull request merged**